Move the registry to in-memory cache

Desvelao opened the pull request:
### Description
This pull request moves the registry to an in-memory cache and removes some logic done by the frontend side that should do the backend side.

#### Changes
- Move the registry management to an in-memory cache managed by ManageHosts
  service.
- Initialize the registry data on the start
  - Update the registry on adding, editing, or deleting an API host
- Move the logic to check if the run_as is enabled to the ManageHosts
  instead of cacheAPIUserAllowRunAs
- Update the backend types of the core plugin
- Remove the process to create/update the registry file when the plugin starts
- Remove the `Installation date` from the About view
- Remove the cacheAPIUserAllowRunAs of the ManageHosts service to use
  the registry data
- Remove the cacheAPIUserAllowRunAs property of Manage hosts
- Remove the UpdateRegistry service
- Remove API endpoint PUT /hosts/update-hostname/{id}
  - Remove API endpoint (route and controller)
  - Remove API request of frontend side
- Remove logic to create the registry file on start
- Remove API endpoint and usage of GET /hosts/remove-orphan-entries
  - Remove API endpoint (route and controller)
  - Remove API request of frontend side
- Remove unused API endpoint GET /api/timestamp
 
### Issues Resolved
#6387

### Evidence
![image](https://github.com/wazuh/wazuh-dashboard-plugins/assets/34042064/c3e30b6c-482a-493f-b275-d63514d9adc1)

### Test
# TEST

With an API connection using an with `run_as` capability enabled with `run_as` disabled, edit the API connection setting `run_as` as `true`, save. The `Run as` column should display the `run_as` is enabled with a `check`

Context: @yenienserrano 

# 6387-registry-in-memory-cache

Legend:
:black_circle:: none
:green_circle:: pass
:yellow_circle:: warning
:red_circle:: fail
:white_circle:: not applicable

## UI

| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
| Go to Server APIs, and add an online API host that can be connected. The data about the cluster name or manager name should appear in the table. | :black_circle: | :black_circle: | :black_circle: |
| Go to Server APIs, and edit an offline API host that can be connected. The data about the cluster name or manager name should appear in the table. | :black_circle: | :black_circle: | :black_circle: |
| Go to Server APIs, and edit an online API host that can not be connected. The data about the cluster name or manager name should not appear in the table. | :black_circle: | :black_circle: | :black_circle: |
| Go to Server APIs, change the cluster name or manager name of an offline API connection, and edit the API connection that can be connected. The new data about the cluster name or manager name should appear in the table. | :black_circle: | :black_circle: | :black_circle: |
| Go to Server APIs, change the cluster name or manager name of an offline API connection, and click on the Refresh button. The new data about the cluster name or manager name should appear in the table. | :black_circle: | :black_circle: | :black_circle: |
| Go to Server APIs, and add an offline API host that can be connected. The data about the cluster name or manager name should not appear in the table. | :black_circle: | :black_circle: | :black_circle: |
| Go to About, it only should display the version and revision of the application. No reference to installation date. | :black_circle: | :black_circle: | :black_circle: |

**Details**
<details>
<summary>:black_circle: Go to Server APIs, and add an online API host that can be connected. The data about the cluster name or manager name should appear in the table.</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Go to Server APIs, and edit an offline API host that can be connected. The data about the cluster name or manager name should appear in the table.</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Go to Server APIs, and edit an online API host that can not be connected. The data about the cluster name or manager name should not appear in the table.</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Go to Server APIs, change the cluster name or manager name of an offline API connection, and edit the API connection that can be connected. The new data about the cluster name or manager name should appear in the table.</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Go to Server APIs, change the cluster name or manager name of an offline API connection, and click on the Refresh button. The new data about the cluster name or manager name should appear in the table.</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Go to Server APIs, and add an offline API host that can be connected. The data about the cluster name or manager name should not appear in the table.</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Go to About, it only should display the version and revision of the application. No reference to installation date.</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>


# 6387-registry-in-memory-cache

Legend:
:black_circle:: none
:green_circle:: pass
:yellow_circle:: warning
:red_circle:: fail
:white_circle:: not applicable

## UI

| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
| Go to Server APIs, and add an online API host that can be connected. The data about the cluster name or manager name should appear in the table. | :black_circle: | :black_circle: | :black_circle: |
| Go to Server APIs, and edit an offline API host that can be connected. The data about the cluster name or manager name should appear in the table. | :black_circle: | :black_circle: | :black_circle: |
| Go to Server APIs, and edit an online API host that can not be connected. The data about the cluster name or manager name should not appear in the table. | :black_circle: | :black_circle: | :black_circle: |
| Go to Server APIs, change the cluster name or manager name of an offline API connection, and edit the API connection that can be connected. The new data about the cluster name or manager name should appear in the table. | :black_circle: | :black_circle: | :black_circle: |
| Go to Server APIs, change the cluster name or manager name of an offline API connection, and click on the Refresh button. The new data about the cluster name or manager name should appear in the table. | :black_circle: | :black_circle: | :black_circle: |
| Go to Server APIs, and add an offline API host that can be connected. The data about the cluster name or manager name should not appear in the table. | :black_circle: | :black_circle: | :black_circle: |
| Go to About, it only should display the version and revision of the application. No reference to installation date. | :black_circle: | :black_circle: | :black_circle: |

**Details**
<details>
<summary>:black_circle: Go to Server APIs, and add an online API host that can be connected. The data about the cluster name or manager name should appear in the table.</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Go to Server APIs, and edit an offline API host that can be connected. The data about the cluster name or manager name should appear in the table.</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Go to Server APIs, and edit an online API host that can not be connected. The data about the cluster name or manager name should not appear in the table.</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Go to Server APIs, change the cluster name or manager name of an offline API connection, and edit the API connection that can be connected. The new data about the cluster name or manager name should appear in the table.</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Go to Server APIs, change the cluster name or manager name of an offline API connection, and click on the Refresh button. The new data about the cluster name or manager name should appear in the table.</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Go to Server APIs, and add an offline API host that can be connected. The data about the cluster name or manager name should not appear in the table.</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Go to About, it only should display the version and revision of the application. No reference to installation date.</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

### Check List
- [ ] All tests pass
  - [ ] `yarn test:jest`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

Desvelao commented:
# TEST

With an API connection using an with run_as` capability enabled with `run_as` disabled, edit the API connection setting `run_as` as `true`, save. The `Run as` column should display the `run_as` is enabled with a `check`

Context: @yenienserrano

yenienserrano commented on plugins/main/server/controllers/wazuh-api.ts:
```diff
@@ -87,6 +84,7 @@ export class WazuhApiCtrl {
           idHost,
         );
       }
+      console.log('HELLO2');
```
Perhaps this can be removed

Desvelao replied:
Removed here: cab4219a27a63e44fd0e77755f8009cd41b46179

yenienserrano commented:
when I delete an API I get an error message but if I press the refresh button the API seems to have been deleted correctly because it does not appear in the table.

<img width="1487" alt="image" src="https://github.com/wazuh/wazuh-dashboard-plugins/assets/63758389/c290acfb-b338-42da-8f68-f9e03e8631d9">

<img width="1480" alt="image" src="https://github.com/wazuh/wazuh-dashboard-plugins/assets/63758389/213b1aee-b820-480f-a223-5ba1a0e08b9f">

Tostti reviewed (changes requested):
## UI

| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
| Go to Server APIs, and add an online API host that can be connected. The data about the cluster name or manager name should appear in the table. | :black_circle: | :green_circle:  | :black_circle: |
| Go to Server APIs, and edit an offline API host that can be connected. The data about the cluster name or manager name should appear in the table. | :black_circle: | :green_circle:  | :black_circle: |
| Go to Server APIs, and edit an online API host that can not be connected. The data about the cluster name or manager name should not appear in the table. | :black_circle: |:green_circle:  | :black_circle: |
| Go to Server APIs, change the cluster name or manager name of an offline API connection, and edit the API connection that can be connected. The new data about the cluster name or manager name should appear in the table. | :black_circle: |:green_circle:  | :black_circle: |
| Go to Server APIs, change the cluster name or manager name of an offline API connection, and click on the Refresh button. The new data about the cluster name or manager name should appear in the table. | :black_circle: | :green_circle: | :black_circle: |
| Go to Server APIs, and add an offline API host that can be connected. The data about the cluster name or manager name should not appear in the table. | :black_circle: | :green_circle:  | :black_circle: |
| Go to About, it only should display the version and revision of the application. No reference to installation date. | :black_circle: | :green_circle: | :black_circle: |
| Changing run as status | :black_circle: | :red_circle: | :black_circle: |
| Removing an API | :black_circle: | :red_circle: | :black_circle: |

**Details**
<details>
<summary>:green_circle: Go to Server APIs, and add an online API host that can be connected. The data about the cluster name or manager name should appear in the table.</summary>

Chrome - :black_circle:

Firefox - :green_circle:
![image](https://github.com/wazuh/wazuh-dashboard-plugins/assets/42900763/701a538b-5c5a-4598-a3bf-91ec9852d2f7)


Safari - :black_circle:

</details>

<details>
<summary>:green_circle: Go to Server APIs, and edit an offline API host that can be connected. The data about the cluster name or manager name should appear in the table.</summary>

Chrome - :black_circle:

Firefox - :green_circle:
![image](https://github.com/wazuh/wazuh-dashboard-plugins/assets/42900763/9a903615-6649-44af-976a-732dc1819fc7)
![image](https://github.com/wazuh/wazuh-dashboard-plugins/assets/42900763/5eed7c77-8161-488c-8c3c-197b4abb9e89)

Safari - :black_circle:

</details>

<details>
<summary>:green_circle: Go to Server APIs, and edit an online API host that can not be connected. The data about the cluster name or manager name should not appear in the table.</summary>

Chrome - :black_circle:

Firefox - :green_circle:
![image](https://github.com/wazuh/wazuh-dashboard-plugins/assets/42900763/14764888-2071-43c5-81e6-39c3f25fefd4)

Safari - :black_circle:

</details>

<details>
<summary>:green_circle: Go to Server APIs, change the cluster name or manager name of an offline API connection, and edit the API connection that can be connected. The new data about the cluster name or manager name should appear in the table.</summary>

Chrome - :black_circle:

Firefox - :green_circle:
![image](https://github.com/wazuh/wazuh-dashboard-plugins/assets/42900763/361e25f1-8d4e-4eaa-aef4-ae9c96074b82)

![image](https://github.com/wazuh/wazuh-dashboard-plugins/assets/42900763/78bc5135-e9ce-477c-a6dc-167f4213d53b)


Safari - :black_circle:

</details>

<details>
<summary>:green_circle: Go to Server APIs, change the cluster name or manager name of an offline API connection, and click on the Refresh button. The new data about the cluster name or manager name should appear in the table.</summary>

Chrome - :black_circle:

Firefox - :green_circle:
![image](https://github.com/wazuh/wazuh-dashboard-plugins/assets/42900763/2f9e5df4-f53d-4a66-a8e1-900f816fd45a)

Safari - :black_circle:

</details>

<details>
<summary>:green_circle: Go to Server APIs, and add an offline API host that can be connected. The data about the cluster name or manager name should not appear in the table.</summary>

Chrome - :black_circle:

Firefox - :green_circle:
![image](https://github.com/wazuh/wazuh-dashboard-plugins/assets/42900763/361e25f1-8d4e-4eaa-aef4-ae9c96074b82)

Safari - :black_circle:

</details>

<details>
<summary>:green_circle: Go to About, it only should display the version and revision of the application. No reference to installation date.</summary>

Chrome - :black_circle:

Firefox - :green_circle:
![image](https://github.com/wazuh/wazuh-dashboard-plugins/assets/42900763/53ae7752-d7d7-4307-9856-852b5646ae8a)

Safari - :black_circle:

</details>


<details>
<summary>🔴  Changing run as status</summary>

Chrome - :black_circle:

Firefox - :red_circle:

Although the change is applied, it is not shown in the table
![image](https://github.com/wazuh/wazuh-dashboard-plugins/assets/42900763/ea177517-9da7-4bcb-aa76-d9b8e87fa173)
![image](https://github.com/wazuh/wazuh-dashboard-plugins/assets/42900763/91d8812d-1c12-4504-a41c-7fd02fa6d266)
![image](https://github.com/wazuh/wazuh-dashboard-plugins/assets/42900763/60b2c4c4-5af7-42fc-bf3d-f9efdc9b5b49)


Safari - :black_circle:

</details>


<details>
<summary>🔴  Removing an API</summary>

Chrome - :black_circle:

Firefox - :red_circle:

If an API is removed (online or offline), this error is shown and the table have to be manually refreshed to show the change
![image](https://github.com/wazuh/wazuh-dashboard-plugins/assets/42900763/bd2bb268-3f40-4db7-81ba-86f9a0410c02)


Safari - :black_circle:

</details>

yenienserrano commented:
when I edit an API I get an error message but if I press the update button the API seems to have been edited correctly. 

<img width="1000" alt="image" src="https://github.com/wazuh/wazuh-dashboard-plugins/assets/63758389/a05c2f5e-4fe7-4fa9-83a6-d481eddd6ad9">
<img width="1000" alt="image" src="https://github.com/wazuh/wazuh-dashboard-plugins/assets/63758389/2061233d-a4c5-40cd-8dca-67c5e96bf11c">

Desvelao commented on plugins/main/server/controllers/wazuh-api.ts:
```diff
@@ -87,6 +84,7 @@ export class WazuhApiCtrl {
           idHost,
         );
       }
+      console.log('HELLO2');
```
Removed here: cab4219a27a63e44fd0e77755f8009cd41b46179

Desvelao commented:
# Changes
- Fixed an error deleting the API host
- Fixed an error when editing an API connection changing the ID. Related to https://github.com/wazuh/wazuh-dashboard-plugins/pull/6481#issuecomment-2004503682
- Fixed an error when enabling the `Run as` and this doesn't display as enabled in the API connection table: Related to https://github.com/wazuh/wazuh-dashboard-plugins/pull/6481#pullrequestreview-1942801334

asteriscos commented:
## UI

| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
| Go to Server APIs, and add an online API host that can be connected. The data about the cluster name or manager name should appear in the table. | :green_circle: | :black_circle: | :black_circle: |
| Go to Server APIs, and edit an offline API host that can be connected. The data about the cluster name or manager name should appear in the table. | :green_circle: | :black_circle: | :black_circle: |
| Go to Server APIs, and edit an online API host that can not be connected. The data about the cluster name or manager name should not appear in the table. | :green_circle: | :black_circle: | :black_circle: |
| Go to Server APIs, change the cluster name or manager name of an offline API connection, and edit the API connection that can be connected. The new data about the cluster name or manager name should appear in the table. | :green_circle: | :black_circle: | :black_circle: |
| Go to Server APIs, change the cluster name or manager name of an offline API connection, and click on the Refresh button. The new data about the cluster name or manager name should appear in the table. | :green_circle: | :black_circle: | :black_circle: |
| Go to Server APIs, and add an offline API host that can be connected. The data about the cluster name or manager name should not appear in the table. | :green_circle: | :black_circle: | :black_circle: |
| Go to About, it only should display the version and revision of the application. No reference to installation date. | :green_circle: | :black_circle: | :black_circle: |

**Details**
<details>
<summary>:green_circle: Go to Server APIs, and add an online API host that can be connected. The data about the cluster name or manager name should appear in the table.</summary>

Chrome - :green_circle:

![image](https://github.com/wazuh/wazuh-dashboard-plugins/assets/9343732/1614773d-2299-4517-a227-223822c24f2e)


Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:green_circle: Go to Server APIs, and edit an offline API host that can be connected. The data about the cluster name or manager name should appear in the table.</summary>

Chrome - :green_circle:

![image](https://github.com/wazuh/wazuh-dashboard-plugins/assets/9343732/18b12458-6792-4e28-b4b6-b7c7cc21ca17)

![image](https://github.com/wazuh/wazuh-dashboard-plugins/assets/9343732/0e43744c-760e-409e-9ec4-287552e4f1b8)


Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:green_circle: Go to Server APIs, and edit an online API host that can not be connected. The data about the cluster name or manager name should not appear in the table.</summary>

Chrome - :green_circle:

![image](https://github.com/wazuh/wazuh-dashboard-plugins/assets/9343732/a77219be-7fb8-4c6f-9231-641afc6de7a0)

![image](https://github.com/wazuh/wazuh-dashboard-plugins/assets/9343732/5ce48462-daf7-4370-a91b-ba8beb276541)


Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:green_circle: Go to Server APIs, change the cluster name or manager name of an offline API connection, and edit the API connection that can be connected. The new data about the cluster name or manager name should appear in the table.</summary>

Chrome - :green_circle:

![image](https://github.com/wazuh/wazuh-dashboard-plugins/assets/9343732/e432a980-af3c-4b47-9f4b-6b01dcee0bbf)

![image](https://github.com/wazuh/wazuh-dashboard-plugins/assets/9343732/9bd7cf3d-d7e2-47d8-b349-fc303c9e651f)


Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:green_circle: Go to Server APIs, change the cluster name or manager name of an offline API connection, and click on the Refresh button. The new data about the cluster name or manager name should appear in the table.</summary>

Chrome - :green_circle:

![image](https://github.com/wazuh/wazuh-dashboard-plugins/assets/9343732/e432a980-af3c-4b47-9f4b-6b01dcee0bbf)

![image](https://github.com/wazuh/wazuh-dashboard-plugins/assets/9343732/9bd7cf3d-d7e2-47d8-b349-fc303c9e651f)

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:green_circle: Go to Server APIs, and add an offline API host that can be connected. The data about the cluster name or manager name should not appear in the table.</summary>

Chrome - :green_circle:

![image](https://github.com/wazuh/wazuh-dashboard-plugins/assets/9343732/1ade2b0e-ed0d-41a3-8eb5-9ecd9fbcac1d)


Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:green_circle: Go to About, it only should display the version and revision of the application. No reference to installation date.</summary>

Chrome - :green_circle:

![image](https://github.com/wazuh/wazuh-dashboard-plugins/assets/9343732/086c4d5e-1513-44fd-ac8e-7ffa94cb2045)


Firefox - :black_circle:

Safari - :black_circle:

</details>
